js/script.js

```diff
@@ -1,8 +1,15 @@
 $(function () {
     $(document).scroll(function () {
-        var $nav = $(".fixed-top");
-        $nav.toggleClass('scrolled', $(this).scrollTop() > $nav.height());
-        var $text = $(".text-white");
-        $text.toggleClass('dark', $(this).scrollTop() > $nav.height())
+        var $nav = $("#nav-bar");
+        var $navSupportedContent = $("#navbarSupportedContent");
+        var isShown = $navSupportedContent.attr("class").includes(" show");
+        if(!isShown)
+            $nav.toggleClass("scrolled", $(this).scrollTop() > $nav.height());
```

Comment:
I don't think the above 3 lines are needed if you make the change I commented below. Makes sense?

Reply:
if this not done it creates a bug, that if menu opened and then scrolled and now menu is closed even if it is scrolled nav bar still turns transparent.

Reply:
Try this code, and let me know if it works perfectly. This makes minimum changes to the code.
```
$(function () {
    var $nav = $("#nav-bar");

    $(document).scroll(function () {
        $nav.toggleClass('scrolled', $(this).scrollTop() > $nav.height());
    });
    $("#toggler").click(() => {
        if ($(this).scrollTop() < $nav.height()) {
            $nav.toggleClass("scrolled");
        }
    });
});
```

Reply:
I found the bug, sorry, wait.

Reply:
this is the same code which creates the above bug.
let's say when page is loaded menu is clicked,
then we scroll it more than the height of navbar then the scroll event handler will make it transparent, which will switch the behavior we were building.
ie transparent when it should be dark, and vise versa

Reply:
and are **minimum changes necessary** for further development on the website?

Reply:
> I found the bug, sorry, wait.

I already wrote that.

> and are minimum changes necessary for further development on the website?

 Sorry what?